src/hal/src/external_memory/mod.rs

```diff
@@ -0,0 +1,263 @@
+//! Structures related to the import external memory functionality
+
+mod errors;
+pub use errors::*;
+
+#[cfg(any(unix))]
+#[cfg_attr(feature = "unstable", doc(cfg(unix)))]
+mod fd;
+#[cfg(any(unix))]
+#[cfg_attr(feature = "unstable", doc(cfg(unix)))]
+pub use fd::*;
+
+#[cfg(any(windows))]
+#[cfg_attr(feature = "unstable", doc(cfg(windows)))]
+mod handle;
+#[cfg(any(windows))]
+#[cfg_attr(feature = "unstable", doc(cfg(windows)))]
+pub use handle::*;
+
+mod ptr;
+pub use ptr::*;
+
+/// Buffer or image
+#[derive(Debug)]
+pub enum BufferOrImage<'a, B: crate::Backend> {
+    /// Buffer
+    Buffer(&'a B::Buffer),
+    /// Image
+    Image(&'a B::Image),
+}
+
+/// External buffer properties
+#[derive(Debug, PartialEq)]
+pub struct ExternalBufferProperties {
+    usage: crate::buffer::Usage,
+    sparse: crate::memory::SparseFlags,
+    external_memory_properties: ExternalMemoryProperties,
+}
+impl ExternalBufferProperties {
+    /// Constructor
+    pub fn new(
+        usage: crate::buffer::Usage,
+        sparse: crate::memory::SparseFlags,
+        external_memory_properties: ExternalMemoryProperties,
+    ) -> Self {
+        Self {
+            usage,
+            sparse,
+            external_memory_properties,
+        }
+    }
+    /// Is the queried configuration exportable
+    pub fn get_queried_buffer_usage(&self) -> crate::buffer::Usage {
```

Comment:
Rust idiomatic style says we don't need `get_` prefix here (true for all accessors)

Reply:
Right, sometime the java ghost of the past come back. I will change that